Docs: Overrides v2 (#2237)
vtex/faststore: apps/site/pages/docs/building-sections/overrides/use-cases/add-an-image-to-your-alert-component.mdx

# Add an image to your Alert component

This guide illustrates an advanced use case for [`getOverriddenSection`](/docs/building-sections/overrides/specification) in FastStore, focusing on customizing the icon of the Alert section to display an image instead.

## Context

- You want to display a custom image alongside the text in your alert messages.
- The native Alert section provides an `icon` prop for customization, but you want more control and flexibility over the displayed content.

> ## Implementation

Reviewer: This comment is still valid. Would be good to have the folder where the section is created here, as we can see in the other docs.

    *
    * Changes the CMS schema to add the src option and remove the icon option.
    * Overrides the Icon option and passes down the other props
    *
    * It's memoized to avoid being re-created every time the AlertWithImage component is re-rendered
    */
    const OverriddenAlert = useMemo(
        () =>
        getOverriddenSection({
            Section: AlertSection,
            components: {
            Icon: { Component: () => <Image src={props.src} alt="custom icon" width={24} height={24} /> },
            },
        }),
        []
    );

    return <OverriddenAlert {...otherProps} icon="noop" />;
    }
   ```

   - This function receives the `AlertWithImageProps`.
   - It destructures the `src` prop containing the image source and remaining props using the spread operator (`...otherProps`).
   - The core functionality lies within the `useMemo` hook.
       - It defines the `OverriddenAlert` constant using `getOverriddenSection`.
       - The override options specify:
           - **`Section`:** The section to be overridden (Alert).
           - **`components`:** An object containing the override for the Icon component.
               - Inside the Icon override:

>                    - **`Component`:** A function returning an `Image` component from the `next/image` library. This displays the image specified by the `src` prop.

Reviewer: ```suggestion
                   - **`Component`:** A function returning an `Image` component. This displays the image specified by the `src` prop.
```

# Add an image to your Alert component

This guide illustrates an advanced use case for [`getOverriddenSection`](/docs/building-sections/overrides/specification) in FastStore. It focuses on customizing the icon of the Alert section to display an image instead. 

> > ℹ️ For detailed instructions on overriding components, refer to [overriding component props](https://developers.vtex.com/docs/guides/faststore/overrides-component-props) and [overriding a component](https://developers.vtex.com/docs/guides/faststore/overrides-native-component).

Reviewer 2: suggestion: Can we change the bg color to call more attention?

> > ℹ️ For detailed instructions on overriding components, refer to [overriding component props](https://developers.vtex.com/docs/guides/faststore/overrides-component-props) and [overriding a component](https://developers.vtex.com/docs/guides/faststore/overrides-native-component).

Reviewer 2: detailes/step-by-step instructions

> 
> > ℹ️ For detailed instructions on overriding components, refer to [overriding component props](https://developers.vtex.com/docs/guides/faststore/overrides-component-props) and [overriding a component](https://developers.vtex.com/docs/guides/faststore/overrides-native-component).

Author: ```suggestion
import { Callout } from 'nextra/components'

<Callout type="info" emoji=" ℹ️ ">
 For detailed instructions on overriding components, refer to [overriding component props](https://developers.vtex.com/docs/guides/faststore/overrides-component-props) and [overriding a component](https://developers.vtex.com/docs/guides/faststore/overrides-native-component).
</Callout>
```

            },
        }),
        []
    );

    return <OverriddenAlert {...otherProps} icon="noop" />;
    }
   ```

   - This function receives the `AlertWithImageProps`.
   - It destructures the `src` prop containing the image source and remaining props using the spread operator (`...otherProps`).
   - The core functionality lies within the `useMemo` hook.
       - It defines the `OverriddenAlert` constant using `getOverriddenSection`.
       - The override options specify:
           - **`Section`:** The section to be overridden (Alert).
           - **`components`:** An object containing the override for the Icon component.
               - Inside the Icon override:
                   - **`Component`:** A function returning an `Image` component. This displays the image specified by the `src` prop.
                   - **`width`** and **`height`:** Optional props to set the size of the image.
       - The use of `useMemo` ensures this override is only created once, improving performance.
   - The function returns the `OverriddenAlert` component, spreading the `otherProps` and setting the `icon` prop to `noop` (effectively disabling the default icon behavior).

> 4. Add the section to the Headless CMS by following the instructions available in [Sending components to the Headless CMS](/docs/building-sections/overrides/sending-components-to-the-headless-cms).

Author: ```suggestion
4. Add the section to the Headless CMS by following the instructions available in [Sending components to the Headless CMS](https://developers.vtex.com/docs/guides/faststore/building-sections-list-of-native-sections).
```

> For the schema it was used the following:

Reviewer 2: I think we need to link, where the people can find the original schema. Which is [here](https://faststore-site-git-docs-override-version-2-faststore.vercel.app/docs/building-sections/overrides/list-of-native-sections#faststore-native-sections)
